File: app/build 15.49.05/tmp/kapt3/stubs/debug/com/anime/art/ai/inject/AppModule.java
package com.anime.art.ai.inject;

@dagger.Module
@kotlin.Metadata(mv = {1, 8, 0}, k = 1, xi = 48, d1 = {"\u0000:\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\b\u0007\u0018\u00002\u00020\u0001B\u0005\u00a2\u0006\u0002\u0010\u0002J\b\u0010\u0003\u001a\u00020\u0004H\u0007J\u0010\u0010\u0005\u001a\u00020\u00062\u0006\u0010\u0007\u001a\u00020\u0004H\u0007J\b\u0010\b\u001a\u00020\tH\u0007J\b\u0010\n\u001a\u00020\u000bH\u0007J\u0010\u0010\f\u001a\u00020\r2\u0006\u0010\u0007\u001a\u00020\u0004H\u0007J\u0010\u0010\u000e\u001a\u00020\u000f2\u0006\u0010\u0007\u001a\u00020\u0004H\u0007J\u0010\u0010\u0010\u001a\u00020\u00112\u0006\u0010\u0012\u001a\u00020\u0006H\u0007\u00a8\u0006\u0013"}, d2 = {"Lcom/anime/art/ai/inject/AppModule;", "", "()V", "provideContext", "Landroid/content/Context;", "provideDatabase", "Lcom/anime/art/ai/data/db/Database;", "context", "provideDezgoApi", "Lcom/anime/art/ai/inject/sinkin/SinkinApi;", "provideFirebaseAnalytic", "Lcom/google/firebase/analytics/FirebaseAnalytics;", "providePreferences", "Lcom/anime/art/ai/data/Preferences;", "providePreferencesConfig", "Lcom/basic/data/PreferencesConfig;", "provideStyleDao", "Lcom/anime/art/ai/data/db/query/StyleDao;", "database", "app_debug"})
@dagger.hilt.InstallIn(value = {dagger.hilt.components.SingletonComponent.class})
public final class AppModule {
    
    public AppModule() {
        super();
    }
    
    @dagger.Provides
    @javax.inject.Singleton
    @org.jetbrains.annotations.NotNull
    public final android.content.Context provideContext() {
        return null;
    }
    
    @dagger.Provides
    @javax.inject.Singleton
    @org.jetbrains.annotations.NotNull
    public final com.google.firebase.analytics.FirebaseAnalytics provideFirebaseAnalytic() {
        return null;
    }
    
    @dagger.Provides
    @javax.inject.Singleton
    @org.jetbrains.annotations.NotNull
    public final com.anime.art.ai.data.Preferences providePreferences(@org.jetbrains.annotations.NotNull
    android.content.Context context) {
        return null;
    }
    
    @dagger.Provides
    @javax.inject.Singleton
    @org.jetbrains.annotations.NotNull
    public final com.basic.data.PreferencesConfig providePreferencesConfig(@org.jetbrains.annotations.NotNull
    android.content.Context context) {
        return null;
    }
    
    @dagger.Provides
    @javax.inject.Singleton
    @org.jetbrains.annotations.NotNull
    public final com.anime.art.ai.inject.sinkin.SinkinApi provideDezgoApi() {
        return null;
    }
    
    @dagger.Provides
    @javax.inject.Singleton
    @org.jetbrains.annotations.NotNull
    public final com.anime.art.ai.data.db.Database provideDatabase(@org.jetbrains.annotations.NotNull
    android.content.Context context) {
        return null;
    }
    
    @dagger.Provides
    @javax.inject.Singleton
    @org.jetbrains.annotations.NotNull
    public final com.anime.art.ai.data.db.query.StyleDao provideStyleDao(@org.jetbrains.annotations.NotNull
    com.anime.art.ai.data.db.Database database) {
        return null;
    }
}

File: app/src/main/java/com/anime/art/ai/inject/AppModule.kt
package com.anime.art.ai.inject

import android.content.Context
import androidx.room.Room
import com.anime.art.ai.common.App
import com.anime.art.ai.common.Constraint
import com.anime.art.ai.data.Preferences
import com.anime.art.ai.data.db.Database
import com.anime.art.ai.data.db.query.GalleryDao
import com.anime.art.ai.data.db.query.PromptDao
import com.anime.art.ai.data.manager.NotificationManagerImpl
import com.anime.art.ai.data.repoository.AIApiRepositoryImpl
import com.anime.art.ai.data.repoository.ServerApiRepositoryImpl
import com.anime.art.ai.data.repoository.SyncRepositoryImpl
import com.anime.art.ai.domain.manager.NotificationManager
import com.anime.art.ai.domain.repository.AIApiRepository
import com.anime.art.ai.domain.repository.ServerApiRepository
import com.anime.art.ai.domain.repository.SyncRepository
import com.anime.art.ai.inject.sinkin.AIApi
import com.anime.art.ai.inject.sinkin.ServerApi
import com.anime.art.ai.inject.sinkin.SinkinApi
import com.basic.data.PreferencesConfig
import com.f2prateek.rx.preferences2.RxSharedPreferences
import com.facebook.stetho.okhttp3.StethoInterceptor
import com.google.firebase.analytics.FirebaseAnalytics
import com.google.firebase.analytics.ktx.analytics
import com.google.firebase.ktx.Firebase
import com.google.gson.GsonBuilder
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import okhttp3.Interceptor
import okhttp3.OkHttpClient
import okhttp3.Response
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.create
import timber.log.Timber
import java.util.concurrent.TimeUnit
import javax.inject.Named
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
class AppModule {

    @Provides
    @Singleton
    fun provideContext(): Context = App.app

    @Provides
    @Singleton
    fun provideFirebaseAnalytic(): FirebaseAnalytics = Firebase.analytics

    @Provides
    @Singleton
    fun providePreferences(context: Context): Preferences {
        val sharedPreferences = context.getSharedPreferences("Preferences", Context.MODE_PRIVATE)
        val rxSharedPreferences = RxSharedPreferences.create(sharedPreferences)
        return Preferences(rxSharedPreferences)
    }

    @Provides
    @Singleton
    fun providePreferencesConfig(context: Context): PreferencesConfig {
        val sharedPreferences = context.getSharedPreferences("PreferencesConfig", Context.MODE_PRIVATE)
        val rxSharedPreferences = RxSharedPreferences.create(sharedPreferences)
        return PreferencesConfig(context, rxSharedPreferences)
    }

    // Server

    @Provides
    @Singleton
    fun provideDezgoApi(): SinkinApi {
        val loggingInterceptor = HttpLoggingInterceptor { message ->
            Timber.d(message)
        }
        loggingInterceptor.level = HttpLoggingInterceptor.Level.BODY

        val networkInterceptor = Interceptor {
            val request = it.request().newBuilder().build()
            it.proceed(request)
        }

        val okHttpClient = OkHttpClient.Builder()
            .addInterceptor { chain ->
                val requestBuilder = chain
                    .request()
                    .newBuilder()

                chain.proceed(requestBuilder.build())
            }
            .addInterceptor(loggingInterceptor)
            .addNetworkInterceptor(networkInterceptor)
            .addNetworkInterceptor(StethoInterceptor())
            .hostnameVerifier { _, _ -> true }
            .retryOnConnectionFailure(false)
            .connectTimeout(1, TimeUnit.MINUTES)
            .writeTimeout(1, TimeUnit.MINUTES)
            .readTimeout(1, TimeUnit.MINUTES)
            .build()

        val gson = GsonBuilder()
            .setLenient()
            .create()

        val retrofit = Retrofit.Builder()
            .baseUrl(Constraint.Sinkin.URL)
            .client(okHttpClient)
            .addConverterFactory(GsonConverterFactory.create(gson))
            .build()

        return retrofit.create(SinkinApi::class.java)
    }
    @Provides
    @Singleton
    @Named("CreditClient")
    fun providesCreditOkHttpClient() : OkHttpClient{
        val loggingInterceptor = HttpLoggingInterceptor { message ->
            Timber.d(message)
        }
        loggingInterceptor.level = HttpLoggingInterceptor.Level.BODY

        val networkInterceptor = Interceptor {
            val request = it.request().newBuilder().build()
            it.proceed(request)
        }

        val okHttpClient = OkHttpClient.Builder()
            .addInterceptor { chain ->
                val requestBuilder = chain
                    .request()
                    .newBuilder()

                chain.proceed(requestBuilder.build())
            }
            .addInterceptor(loggingInterceptor)
            .addNetworkInterceptor(networkInterceptor)
            .addNetworkInterceptor(StethoInterceptor())
            .hostnameVerifier { _, _ -> true }
            .retryOnConnectionFailure(false)
            .connectTimeout(1, TimeUnit.MINUTES)
            .writeTimeout(1, TimeUnit.MINUTES)
            .readTimeout(1, TimeUnit.MINUTES)
            .build()

        return okHttpClient
    }
    @Provides
    @Singleton
    @Named("AIClient")
    fun providesAIOkHttpClient() : OkHttpClient{
        val loggingInterceptor = HttpLoggingInterceptor { message ->
            Timber.d(message)
        }
        loggingInterceptor.level = HttpLoggingInterceptor.Level.BODY

        val networkInterceptor = Interceptor {
            val request = it.request().newBuilder().build()
            it.proceed(request)
        }

        val okHttpClient = OkHttpClient.Builder()
            .addInterceptor(object  : Interceptor{
                override fun intercept(chain: Interceptor.Chain): Response {
                    val originalRequest = chain.request()
                    val authenticatedRequest = originalRequest.newBuilder()
                        .header("Authorization", "Bearer ${Constraint.AIGeneration.KEY}")
                        .build()
                    return chain.proceed(authenticatedRequest)
                }

            })
            .addInterceptor { chain ->
                val requestBuilder = chain
                    .request()
                    .newBuilder()

                chain.proceed(requestBuilder.build())
            }
            .addInterceptor(loggingInterceptor)
            .addNetworkInterceptor(networkInterceptor)
            .addNetworkInterceptor(StethoInterceptor())
            .hostnameVerifier { _, _ -> true }
            .retryOnConnectionFailure(false)
            .connectTimeout(1, TimeUnit.MINUTES)
            .writeTimeout(1, TimeUnit.MINUTES)
            .readTimeout(1, TimeUnit.MINUTES)
            .build()

        return okHttpClient
    }

    @Provides
    @Singleton
    fun providesGsonConverterFactory() : GsonConverterFactory{
        val gson = GsonBuilder().setLenient().create()
        return GsonConverterFactory.create(gson)
    }

    @Provides
    @Singleton
    @Named("CreditServer")
    fun providesRetrofitCreditServer(gsonConverterFactory: GsonConverterFactory,
    @Named("CreditClient") okHttpClient: OkHttpClient) : Retrofit{
        return Retrofit.Builder()
            .addConverterFactory(gsonConverterFactory)
            .baseUrl(Constraint.BASE_URL)
            .client(okHttpClient)
            .build()
    }
    @Provides
    @Singleton
    @Named("AIServer")
    fun providesRetrofitAIServer(gsonConverterFactory: GsonConverterFactory,
                                 @Named("AIClient") okHttpClient: OkHttpClient) : Retrofit{
        return Retrofit.Builder()
            .addConverterFactory(gsonConverterFactory)
            .baseUrl(Constraint.AIGeneration.URL)
            .client(okHttpClient)
            .build()
    }
    @Provides
    fun providesServerApi(@Named("CreditServer") retrofit: Retrofit) : ServerApi{
        return retrofit.create(ServerApi::class.java)
    }

    @Provides
    fun providesAIApi(@Named("AIServer") retrofit: Retrofit) : AIApi{
        return retrofit.create(AIApi::class.java)
    }
    // Database

    @Provides
    @Singleton
    fun provideDatabase(context: Context): Database = Room.databaseBuilder(
        context,
        Database::class.java,
        Database.DB_NAME
    )
        .fallbackToDestructiveMigration()
        .allowMainThreadQueries()
        .build()

    @Provides
    @Singleton
    fun provideStyleDao(database: Database): GalleryDao = database.galleryDao()

    @Provides
    @Singleton
    fun providesPromptDao(database: Database) : PromptDao = database.promptDao()
    // Repository

    @Provides
    @Singleton
    fun provideSyncRepositoryImpl(repo: SyncRepositoryImpl): SyncRepository = repo

    @Provides
    @Singleton
    fun provideServerApiRepositoryImpl(repo: ServerApiRepositoryImpl): ServerApiRepository = repo

    @Provides
    @Singleton
    fun providesAIApiRepositoryImpl(repo: AIApiRepositoryImpl): AIApiRepository = repo
    // Manager

    @Provides
    @Singleton
    fun provideNotificationManagerImpl(manager: NotificationManagerImpl): NotificationManager = manager

}